docs/core/event_handler/bedrock_agents.md

```diff
@@ -0,0 +1,330 @@
+You can print this string or save it to a file. You'll use the file later when creating the Agent.
+
+You'll need to regenerate the OpenAPI schema and update your Agent everytime your API changes.
+
+=== "app.py"
+
+    ```python hl_lines="24 25"
+    --8<-- "examples/event_handler_bedrock_agents/src/generating_openapi_schema.py"
+    ```
+
+	1. This ensures that it's only executed when running the file directly, and not when running on the Lambda runtime.
+    2. You can use [additional options](#customizing-openapi-metadata) to customize the OpenAPI schema.
+
+=== "OpenAPI schema"
+
+    ```json hl_lines="13 16 24"
+    --8<-- "examples/event_handler_bedrock_agents/src/generating_openapi_schema.json"
+    ```
+
+To get the OpenAPI schema, run the Python script from your terminal.
+The script will generate the schema directly to standard output, which you can redirect to a file.
+
+```sh
+python3 app.py > schema.json
+```
+
+### Crafting effective OpenAPI schemas
+
+Working with Agents for Amazon Bedrock will introduce [non-deterministic behaviour to your system](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-how.html#agents-rt){target="_blank"}.
+
+???+ note "Why is that?"
+	Amazon Bedrock uses LLMs to understand and respond to user input.
+	These models are trained on vast amounts of data and are capable of extracting meanings from a sequence of text and understanding the relationship between words and phrases on it.
+	However, this means that the same input can result in different outputs, depending on the characteristics of the LLM being used.
+
+The OpenAPI schema provides context and semantics to the Agent that will support the decision process for invoking our Lambda function.
+Sparse or ambiguous schemas can result in unexpected outcomes.
+
+We recommend enriching your OpenAPI schema with as many details as possible to help the Agent understand your functions, and make correct invocations.
+To achieve that, keep the following suggestions in mind:
+
+* Always describe your function behaviour using the `description` field in your annotations
+* When refactoring, update your description field to match the function outcomes
+* Use distinct `description` for each function to have clear separation of semantics
+
+### Video walkthrough
+
+To create an Agent for Amazon Bedrock, refer to the [official documentation](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-create.html) provided by AWS.
+
+The following video demonstrates the end-to-end process:
+
+<center>
+<iframe width="720" height="405" src="https://www.youtube-nocookie.com/embed/NWoC5FTSt7s?si=AG2qpLJbxCkyiLma&amp;controls=1" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+</center>
+
+During the creation process, you should use the schema generated in the previous step when prompted for an OpenAPI specification.
```

Comment:
Instead of stating "previous step", create a link that goes to the relevant docs section